File: app/src/main/java/com/veganbeauty/app/features/account/notification/AccountNotificationFragment.java
package com.veganbeauty.app.features.account.notification;

import android.Manifest;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.lifecycle.ViewModelProvider;

import com.veganbeauty.app.R;
import com.veganbeauty.app.core.base.RootieFragment;
import com.veganbeauty.app.data.local.LocalJsonReader;
import com.veganbeauty.app.data.local.ProfileSession;
import com.veganbeauty.app.data.local.entities.BookingHistoryEntity;
import com.veganbeauty.app.data.local.entities.NotificationItem;
import com.veganbeauty.app.data.repository.NotificationRepository;
import com.veganbeauty.app.databinding.AccountNotificationFragmentBinding;
import com.veganbeauty.app.features.account.checkin.AccountCheckinFragment;
import com.veganbeauty.app.features.account.expiry.AccountProductExpiryFragment;
import com.veganbeauty.app.features.account.order.AccountOrderDetailFragment;
import com.veganbeauty.app.features.myskin.AccountSyncHelper;
import com.veganbeauty.app.features.myskin.BookingDetailCancelledFragment;
import com.veganbeauty.app.features.myskin.BookingDetailCompletedFragment;
import com.veganbeauty.app.features.myskin.BookingDetailUpcomingFragment;
import com.veganbeauty.app.features.profile.AccountVoucherDetailFragment;
import com.veganbeauty.app.features.profile.VoucherListAdapter.VoucherItem;
import com.veganbeauty.app.features.routine.SkinReminderFragment;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import kotlinx.coroutines.flow.FlowCollector;

public class AccountNotificationFragment extends RootieFragment {

    private AccountNotificationFragmentBinding _binding;
    private NotificationViewModel viewModel;
    private NotificationListAdapter listAdapter;

    @Override
    protected boolean shouldSetupNotificationBell() {
        return false;
    }

    private final ActivityResultLauncher<String> requestPermissionLauncher = registerForActivityResult(
            new ActivityResultContracts.RequestPermission(),
            isGranted -> {
                if (isGranted) {
                    Toast.makeText(requireContext(), "Đã cấp quyền nhận thông báo", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(requireContext(), "Bạn chưa cấp quyền nhận thông báo", Toast.LENGTH_SHORT).show();
                }
            }
    );

    private AccountNotificationFragmentBinding getBinding() {
        return _binding;
    }

    private void checkNotificationPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            if (ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.POST_NOTIFICATIONS)
                    != PackageManager.PERMISSION_GRANTED) {
                requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS);
            }
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        _binding = AccountNotificationFragmentBinding.inflate(inflater, container, false);
        
        listAdapter = new NotificationListAdapter(
                item -> {
                    viewModel.markAsRead(item.getId());
                    handleNotificationNavigation(item);
                },
                item -> {
                    viewModel.markAsRead(item.getId());
                    if ("COPY MÃ".equals(item.getActionText()) && item.getVoucherCode() != null && !item.getVoucherCode().isEmpty()) {
                        if (getContext() != null) {
                            copyToClipboard(getContext(), item.getVoucherCode());
                            Toast.makeText(getContext(), "Đã sao chép mã voucher " + item.getVoucherCode() + " thành công!", Toast.LENGTH_SHORT).show();
                        }
                    }
                    handleNotificationNavigation(item);
                },
                item -> {
                    viewModel.markAsRead(item.getId());
                },
                item -> {
                    viewModel.deleteNotification(item.getId());
                }
        );

        setupViewModel();
        return getBinding().getRoot();
    }

    private void setupViewModel() {
        NotificationRepository repository = NotificationRepository.getInstance(requireContext());
        viewModel = new ViewModelProvider(this, new NotificationViewModelFactory(repository)).get(NotificationViewModel.class);
    }

    @Override
    public void setupUI(@NonNull View view) {
        checkNotificationPermission();
        
        getBinding().btnBack.setOnClickListener(v -> {
            if (getActivity() != null) {
                getActivity().getOnBackPressedDispatcher().onBackPressed();
            }
        });

        getBinding().rvNotifications.setAdapter(listAdapter);

        getBinding().tabAll.setOnClickListener(v -> viewModel.selectTab("Tất cả"));
        getBinding().tabPromo.setOnClickListener(v -> viewModel.selectTab("Khuyến mãi"));
        getBinding().tabOrder.setOnClickListener(v -> viewModel.selectTab("Đơn hàng"));
        getBinding().tabOther.setOnClickListener(v -> viewModel.selectTab("Khác"));

        getBinding().etSearch.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {}
            @Override
            public void afterTextChanged(Editable s) {
                viewModel.setSearchQuery(s != null ? s.toString() : "");
            }
        });

        getBinding().btnMarkAllRead.setOnClickListener(v -> {
            viewModel.markAllAsRead();
            Toast.makeText(getContext(), "Đã đánh dấu đọc tất cả", Toast.LENGTH_SHORT).show();
        });

        getBinding().btnDeleteAll.setOnClickListener(v -> {
            new androidx.appcompat.app.AlertDialog.Builder(requireContext())
                    .setTitle("Xoá tất cả thông báo")
                    .setMessage("Bạn có chắc chắn muốn xoá toàn bộ thông báo không?")
                    .setPositiveButton("Xoá", (dialog, which) -> {
                        viewModel.deleteAllNotifications();
                        Toast.makeText(getContext(), "Đã xoá toàn bộ thông báo", Toast.LENGTH_SHORT).show();
                    })
                    .setNegativeButton("Hủy", null)
                    .show();
        });
    }

    @Override
    public void observeViewModel() {
        viewModel.notificationItems.observe(getViewLifecycleOwner(), items -> {
            listAdapter.submitList(items);
            boolean isEmpty = items.isEmpty();
            getBinding().layoutEmptyState.setVisibility(isEmpty ? View.VISIBLE : View.GONE);
            getBinding().rvNotifications.setVisibility(isEmpty ? View.GONE : View.VISIBLE);
            getBinding().layoutHeaderActions.setVisibility(isEmpty ? View.GONE : View.VISIBLE);
        });

        viewModel.selectedTab.observe(getViewLifecycleOwner(), this::updateTabStyles);
    }

    private void updateTabStyles(String activeTab) {
        Map<String, TextView> tabs = new HashMap<>();
        tabs.put("Tất cả", getBinding().tabAll);
        tabs.put("Khuyến mãi", getBinding().tabPromo);
        tabs.put("Đơn hàng", getBinding().tabOrder);
        tabs.put("Khác", getBinding().tabOther);

        for (Map.Entry<String, TextView> entry : tabs.entrySet()) {
            TextView textView = entry.getValue();
            if (entry.getKey().equalsIgnoreCase(activeTab)) {
                textView.setBackgroundResource(R.drawable.bg_btn_buy);
                textView.setTextColor(ContextCompat.getColor(requireContext(), R.color.white));
            } else {
                textView.setBackgroundResource(R.drawable.tab_inactive_bg);
                textView.setTextColor(ContextCompat.getColor(requireContext(), R.color.primary));
            }
        }
    }

    private void handleNotificationNavigation(NotificationItem item) {
        String type = (item.getNotificationType() != null ? item.getNotificationType() : "").toLowerCase();
        String category = (item.getCategory() != null ? item.getCategory() : "").toLowerCase();
        String title = (item.getTitle() != null ? item.getTitle() : "").toLowerCase();
        String content = (item.getContent() != null ? item.getContent() : "").toLowerCase();

        if (type.equals("voucher") || category.contains("khuyến mãi") || title.contains("voucher") || content.contains("voucher")) {
            String code = item.getVoucherCode() != null ? item.getVoucherCode() : "RT50KDEC";
            VoucherItem voucher = NotificationIntentHandler.findVoucherByCode(requireContext(), code);
            if (voucher == null) {
                voucher = new VoucherItem(
                        item.getId(),
                        item.getTitle(),
                        item.getContent(),
                        code,
                        "valid",
                        "2026-12-31 23:59:59",
                        "discount",
                        false,
                        1,
                        300000,
                        "Tất cả sản phẩm",
                        "fixed_amount",
                        50000
                );
            }
            com.veganbeauty.app.features.profile.VoucherListAdapter.VoucherItem mappedVoucher = new com.veganbeauty.app.features.profile.VoucherListAdapter.VoucherItem(
                    voucher.getId(), voucher.getTitle(), voucher.getDescription(), voucher.getCode(),
                    voucher.getStatus(), voucher.getHsd(), voucher.getType(), voucher.isFromGift(),
                    voucher.getQuantity(), voucher.getMinOrderValue(), voucher.getApplicableProducts(),
                    voucher.getOfferType(), voucher.getDiscountValue()
            );
            AccountVoucherDetailFragment fragment = AccountVoucherDetailFragment.newInstance(mappedVoucher);
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();

        } else if (type.equals("order") || category.contains("đơn hàng") || title.contains("đơn hàng") || content.contains("đơn hàng")) {
            String orderId = item.getOrderId() != null ? item.getOrderId() : "RT8829";
            AccountOrderDetailFragment fragment = AccountOrderDetailFragment.newInstance(orderId);
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();

        } else if (type.equals("checkin") || title.contains("điểm danh") || content.contains("điểm danh")) {
            AccountCheckinFragment fragment = new AccountCheckinFragment();
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();

        } else if (type.equals("product expired") || title.contains("hết hạn") || content.contains("hết hạn")) {
            AccountProductExpiryFragment fragment = new AccountProductExpiryFragment();
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();

        } else if (type.equals("schedule date") || title.contains("lịch hẹn") || content.contains("lịch hẹn")) {
            String scheduleId = item.getScheduleId() != null ? item.getScheduleId() : "BK_NOTI_101";
            String userEmail = ProfileSession.getEmail(requireContext());
            List<BookingHistoryEntity> bookings = new LocalJsonReader(requireContext()).getUserBookingHistory(userEmail);
            
            BookingHistoryEntity realBooking = null;
            for (BookingHistoryEntity b : bookings) {
                if (b.getId().equals(scheduleId)) {
                    realBooking = b;
                    break;
                }
            }

            RootieFragment fragment;
            if (realBooking != null) {
                switch (realBooking.getStatus()) {
                    case "Đã hoàn thành":
                        fragment = BookingDetailCompletedFragment.newInstance(realBooking);
                        break;
                    case "Đã huỷ":
                        fragment = BookingDetailCancelledFragment.newInstance(realBooking);
                        break;
                    default:
                        fragment = BookingDetailUpcomingFragment.newInstance(realBooking);
                        break;
                }
            } else {
                BookingHistoryEntity mockBooking = new BookingHistoryEntity(
                        scheduleId, "user_1", "Nguyễn Khánh Xuân", "0901234567", userEmail,
                        "Chăm sóc da chuyên sâu Acne Free", "15 Tháng 6, 2026", "Thứ Hai",
                        "14:00 - 15:30", "90 phút", "Rootie Quận 1", "123 Nguyễn Thị Minh Khai, Quận 1, TP. HCM",
                        "Sắp diễn ra"
                );
                fragment = BookingDetailUpcomingFragment.newInstance(mockBooking);
            }
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();

        } else if (type.equals("skin care") || title.contains("dưỡng da") || title.contains("chăm sóc da") || content.contains("dưỡng da") || content.contains("chăm sóc da")) {
            SkinReminderFragment fragment = new SkinReminderFragment();
            getParentFragmentManager().beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit();
        }
    }

    private void copyToClipboard(Context context, String text) {
        ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clip = ClipData.newPlainText("Rootie Voucher Code", text);
        clipboard.setPrimaryClip(clip);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        _binding = null;
    }

    @Override
    public void onResume() {
        super.onResume();
        AccountSyncHelper.sync(requireContext(), () -> {
            if (getActivity() != null) {
                getActivity().runOnUiThread(() ->
                        NotificationRepository.getInstance(requireContext()).refreshNotifications()
                );
            }
        });
    }
}

File: app/src/main/java/com/veganbeauty/app/features/account/notification/AccountNotificationFragment.kt
package com.veganbeauty.app.features.account.notification

import android.Manifest
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.content.ContextCompat
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.veganbeauty.app.R
import com.veganbeauty.app.core.base.RootieFragment
import com.veganbeauty.app.data.local.entities.NotificationItem
import com.veganbeauty.app.data.repository.NotificationRepository
import com.veganbeauty.app.databinding.AccountNotificationFragmentBinding
import com.veganbeauty.app.data.local.LocalJsonReader
import com.veganbeauty.app.data.local.entities.BookingHistoryEntity
import com.veganbeauty.app.features.profile.AccountVoucherDetailFragment
import com.veganbeauty.app.features.profile.VoucherItem
import com.veganbeauty.app.features.account.order.AccountOrderDetailFragment
import com.veganbeauty.app.features.account.checkin.AccountCheckinFragment
import com.veganbeauty.app.features.account.expiry.AccountProductExpiryFragment
import com.veganbeauty.app.features.myskin.BookingHistoryFragment
import com.veganbeauty.app.features.myskin.BookingDetailUpcomingFragment
import com.veganbeauty.app.features.myskin.BookingDetailCompletedFragment
import com.veganbeauty.app.features.myskin.BookingDetailCancelledFragment
import com.veganbeauty.app.features.routine.SkinReminderFragment

class AccountNotificationFragment : RootieFragment() {

    private val requestPermissionLauncher = registerForActivityResult(
        ActivityResultContracts.RequestPermission()
    ) { isGranted: Boolean ->
        if (isGranted) {
            Toast.makeText(requireContext(), "Đã cấp quyền nhận thông báo", Toast.LENGTH_SHORT).show()
        } else {
            Toast.makeText(requireContext(), "Bạn chưa cấp quyền nhận thông báo", Toast.LENGTH_SHORT).show()
        }
    }

    private fun checkNotificationPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            if (ContextCompat.checkSelfPermission(
                    requireContext(),
                    Manifest.permission.POST_NOTIFICATIONS
                ) != PackageManager.PERMISSION_GRANTED
            ) {
                requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
            }
        }
    }

    private var _binding: AccountNotificationFragmentBinding? = null
    private val binding get() = _binding!!

    private lateinit var viewModel: NotificationViewModel

    private val listAdapter by lazy {
        NotificationListAdapter(
            onItemClick = { item ->
                viewModel.markAsRead(item.id)
                handleNotificationNavigation(item)
            },
            onActionClick = { item ->
                viewModel.markAsRead(item.id)
                if (item.actionText == "COPY MÃ" && !item.voucherCode.isNullOrEmpty()) {
                    context?.let { ctx ->
                        copyToClipboard(ctx, item.voucherCode)
                        Toast.makeText(
                            ctx,
                            "Đã sao chép mã voucher ${item.voucherCode} thành công!",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
                handleNotificationNavigation(item)
            },
            onMarkReadClick = { item ->
                viewModel.markAsRead(item.id)
            },
            onDeleteClick = { item ->
                viewModel.deleteNotification(item.id)
            }
        )
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = AccountNotificationFragmentBinding.inflate(inflater, container, false)
        setupViewModel()
        return binding.root
    }

    private fun setupViewModel() {
        val repository = NotificationRepository.getInstance(requireContext())
        viewModel = ViewModelProvider(this, object : ViewModelProvider.Factory {
            override fun <T : ViewModel> create(modelClass: Class<T>): T {
                @Suppress("UNCHECKED_CAST")
                return NotificationViewModel(repository) as T
            }
        })[NotificationViewModel::class.java]
    }

    override fun setupUI(view: View) {
        checkNotificationPermission()
        // Back Navigation
        binding.btnBack.setOnClickListener {
            activity?.onBackPressedDispatcher?.onBackPressed()
        }

        // Bind RecyclerView
        binding.rvNotifications.adapter = listAdapter

        // Bind Chips/Tabs Clicks
        binding.tabAll.setOnClickListener { viewModel.selectTab("Tất cả") }
        binding.tabPromo.setOnClickListener { viewModel.selectTab("Khuyến mãi") }
        binding.tabOrder.setOnClickListener { viewModel.selectTab("Đơn hàng") }
        binding.tabOther.setOnClickListener { viewModel.selectTab("Khác") }

        // Bind Search Input Bar
        binding.etSearch.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable?) {
                viewModel.setSearchQuery(s?.toString().orEmpty())
            }
        })

        // Bind Header Bulk Actions
        binding.btnMarkAllRead.setOnClickListener {
            viewModel.markAllAsRead()
            Toast.makeText(context, "Đã đánh dấu đọc tất cả", Toast.LENGTH_SHORT).show()
        }

        binding.btnDeleteAll.setOnClickListener {
            androidx.appcompat.app.AlertDialog.Builder(requireContext())
                .setTitle("Xoá tất cả thông báo")
                .setMessage("Bạn có chắc chắn muốn xoá toàn bộ thông báo không?")
                .setPositiveButton("Xoá") { _, _ ->
                    viewModel.deleteAllNotifications()
                    Toast.makeText(context, "Đã xoá toàn bộ thông báo", Toast.LENGTH_SHORT).show()
                }
                .setNegativeButton("Hủy", null)
                .show()
        }
    }

    override fun observeViewModel() {
        // Observe search and tab filters to render lists
        viewModel.notificationItems.observe(viewLifecycleOwner) { items ->
            listAdapter.submitList(items)
            val isEmpty = items.isEmpty()
            binding.layoutEmptyState.visibility = if (isEmpty) View.VISIBLE else View.GONE
            binding.rvNotifications.visibility = if (isEmpty) View.GONE else View.VISIBLE
            binding.layoutHeaderActions.visibility = if (isEmpty) View.GONE else View.VISIBLE
        }

        // Observe active tab to update visual styles dynamically
        viewModel.selectedTab.observe(viewLifecycleOwner) { activeTab ->
            updateTabStyles(activeTab)
        }
    }

    private fun updateTabStyles(activeTab: String) {
        val tabs = mapOf(
            "Tất cả" to binding.tabAll,
            "Khuyến mãi" to binding.tabPromo,
            "Đơn hàng" to binding.tabOrder,
            "Khác" to binding.tabOther
        )

        for ((tab, textView) in tabs) {
            if (tab.equals(activeTab, ignoreCase = true)) {
                textView.setBackgroundResource(R.drawable.tab_active_bg)
                textView.setTextColor(ContextCompat.getColor(requireContext(), R.color.white))
            } else {
                textView.setBackgroundResource(R.drawable.tab_inactive_bg)
                textView.setTextColor(ContextCompat.getColor(requireContext(), R.color.primary))
            }
        }
    }

    private fun handleNotificationNavigation(item: NotificationItem) {
        val type = (item.notificationType ?: "").lowercase()
        val category = (item.category ?: "").lowercase()
        val title = (item.title ?: "").lowercase()
        val content = (item.content ?: "").lowercase()

        when {
            type == "voucher" || category.contains("khuyến mãi") || title.contains("voucher") || content.contains("voucher") -> {
                val code = item.voucherCode ?: "RT50KDEC"
                val voucher = NotificationIntentHandler.findVoucherByCode(requireContext(), code) ?: VoucherItem(
                    id = item.id,
                    title = item.title,
                    description = item.content,
                    code = code,
                    status = "valid",
                    hsd = "2026-12-31 23:59:59",
                    type = "discount",
                    fromGift = false,
                    quantity = 1,
                    minOrderValue = 300000,
                    applicableProducts = "Tất cả sản phẩm",
                    offerType = "fixed_amount",
                    discountValue = 50000
                )
                val fragment = AccountVoucherDetailFragment.newInstance(voucher)
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
            type == "order" || category.contains("đơn hàng") || title.contains("đơn hàng") || content.contains("đơn hàng") -> {
                val orderId = item.orderId ?: "RT8829"
                val fragment = AccountOrderDetailFragment.newInstance(orderId)
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
            type == "checkin" || title.contains("điểm danh") || content.contains("điểm danh") -> {
                val fragment = AccountCheckinFragment()
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
            type == "product expired" || title.contains("hết hạn") || content.contains("hết hạn") -> {
                val fragment = AccountProductExpiryFragment()
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
            type == "schedule date" || title.contains("lịch hẹn") || content.contains("lịch hẹn") -> {
                val scheduleId = item.scheduleId ?: "BK_NOTI_101"
                val userEmail = com.veganbeauty.app.data.local.ProfileSession.getEmail(requireContext())
                val bookings = LocalJsonReader(requireContext()).getUserBookingHistory(userEmail)
                val realBooking = bookings.find { it.id == scheduleId }
                val fragment = if (realBooking != null) {
                    when (realBooking.status) {
                        "Đã hoàn thành" -> BookingDetailCompletedFragment.newInstance(realBooking)
                        "Đã huỷ" -> BookingDetailCancelledFragment.newInstance(realBooking)
                        else -> BookingDetailUpcomingFragment.newInstance(realBooking)
                    }
                } else {
                    val mockBooking = BookingHistoryEntity(
                        id = scheduleId,
                        userId = "user_1",
                        userName = "Nguyễn Khánh Xuân",
                        userPhone = "0901234567",
                        userEmail = userEmail,
                        serviceName = "Chăm sóc da chuyên sâu Acne Free",
                        dateDisplay = "15 Tháng 6, 2026",
                        dayOfWeek = "Thứ Hai",
                        time = "14:00 - 15:30",
                        duration = "90 phút",
                        storeName = "Rootie Quận 1",
                        storeAddress = "123 Nguyễn Thị Minh Khai, Quận 1, TP. HCM",
                        status = "Sắp diễn ra"
                    )
                    BookingDetailUpcomingFragment.newInstance(mockBooking)
                }
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
            type == "skin care" || title.contains("dưỡng da") || title.contains("chăm sóc da") || content.contains("dưỡng da") || content.contains("chăm sóc da") -> {
                val fragment = SkinReminderFragment()
                parentFragmentManager.beginTransaction()
                    .replace(R.id.main_container, fragment)
                    .addToBackStack(null)
                    .commit()
            }
        }
    }

    private fun copyToClipboard(context: Context, text: String) {
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText("Rootie Voucher Code", text)
        clipboard.setPrimaryClip(clip)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
